numba/parfor.py

```diff
@@ -1010,52 +970,52 @@ def _gen_np_divide(arg1, arg2, out_ir, typemap):
 
 
 def _gen_arrayexpr_getitem(
+        func_ir,
         var,
+        size_vars,
         parfor_index_tuple_var,
         all_parfor_indices,
         el_typ,
         calltypes,
         typemap,
-        array_shape_classes,
         out_ir):
     """if there is implicit dimension broadcast, generate proper access variable
-    for getitem. For example, if indices are (i1,i2,i3) but shape correlations
-    are (c1,0,c3), generate a tuple with (i1,0,i3) for access.
-    Another example: for (i1,i2,i3) and (c1,c2) generate (i2,i3).
+    for getitem. For example, if indices are (i1,i2,i3) but shape is (c1,0,c3),
+    generate a tuple with (i1,0,i3) for access.  Another example: for (i1,i2,i3)
+    and (c1,c2) generate (i2,i3).
     """
     loc = var.loc
     index_var = parfor_index_tuple_var
-    shape_corrs = copy.copy(array_shape_classes[var.name])
     ndims = typemap[var.name].ndim
     num_indices = len(all_parfor_indices)
-    if 0 in shape_corrs or ndims < num_indices:
-        if ndims == 1:
-            # broadcast prepends dimensions, so use last index for 1D arrays
-            index_var = all_parfor_indices[-1]
-        else:
-            # broadcast prepends dimensions so ignore indices from beginning
-            ind_offset = num_indices - ndims
-            tuple_var = ir.Var(var.scope, mk_unique_var(
-                "$parfor_index_tuple_var_bcast"), loc)
-            typemap[tuple_var.name] = types.containers.UniTuple(types.intp,
-                                                                ndims)
-            # const var for size 1 dim access index: $const0 = Const(0)
-            const_node = ir.Const(0, var.loc)
-            const_var = ir.Var(var.scope, mk_unique_var("$const_ind_0"), loc)
-            typemap[const_var.name] = types.intp
-            const_assign = ir.Assign(const_node, const_var, loc)
-            out_ir.append(const_assign)
-            index_vars = []
-            for i in reversed(range(ndims)):
-                if shape_corrs[i] == 0:
-                    index_vars.append(const_var)
-                else:
-                    index_vars.append(all_parfor_indices[ind_offset + i])
-            index_vars = list(reversed(index_vars))
-            tuple_call = ir.Expr.build_tuple(index_vars, loc)
-            tuple_assign = ir.Assign(tuple_call, tuple_var, loc)
-            out_ir.append(tuple_assign)
-            index_var = tuple_var
+    if ndims == 1:
```

Comment:
Thanks for point it out. It is addressed by 76f0ecdd9ebd0fa9b1fa6f7cdce13dc59608c922